### PR TITLE
User can logout

### PR DESCRIPTION
**Pull request by julieahlquist**
### User Story

```
As a user,
In order to protect my personal information,
I want to be able to log out after I have logged in.
```
### Feature description
Link to this [feature on Pivotal Tracker](https://www.pivotaltracker.com/story/show/168106340)
### Description of PR
Enables the user to log out by clicking log-out button displayed in navbar when user is logged in. After successful logout a "welcome back"-flash is displayed. 

### Screenshots
http://g.recordit.co/pzbarsT7a1.gif

### What did we learn?
Gaining knowledge of `signOutUser` methods of Redux Token Auth.

### CODE REVIEW

File: cypress/integration/userCanLogout.spec.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    cy.route({
+      method: 'DELETE',
+      url: 'http://localhost:3000/api/v0/auth/sign_out',
+      response: 'fixture:successful_user_logout.json',
+    })
+    cy.visit("http://localhost:3001")
+    cy.university_login("harvard@mail.com", "password");
+    cy.get("#logout-button").should("exist");
+    cy.get("#logout-button").click();
+    cy.get("#flash").should("contain", "Successful logout. Welcome back soon!")
```
Comment (ghost) [inline cypress/integration/userCanLogout.spec.js]: I think the phrase you're looking for is "Come back soon", but for the PR, I would just leave it at "You have successfully logged out."

File: cypress/integration/userCanLogout.spec.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+describe('User Log-Out', () => {
+   beforeEach(() => {
```
Comment (ghost) [inline cypress/integration/userCanLogout.spec.js:2]: You don't have to GET articles to test this feature

File: cypress/integration/userCanLogout.spec.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  it("Can log out successfully", () => {
+    cy.route({
+      method: 'DELETE',
+      url: 'http://localhost:3000/api/v0/auth/sign_out',
+      response: 'fixture:successful_user_logout.json',
+    })
+    cy.visit("http://localhost:3001")
+    cy.university_login("harvard@mail.com", "password");
+    cy.get("#logout-button").should("exist");
```
Comment (ghost) [inline cypress/integration/userCanLogout.spec.js:28]: There would be value in testing each type of account, not just university account.

File: cypress/integration/userCanLogout.spec.js (hunk 1 above)
Comment (julieahlquist) [reply cypress/integration/userCanLogout.spec.js]: thanks, fixed

File: cypress/fixtures/successful_user_logout.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "success": true
```
Comment (ghost) [inline cypress/fixtures/successful_user_logout.json:2]: Is this all that will be sent from the back end after a logout action?

File: cypress/integration/userCanLogout.spec.js (hunk 3 above)
Comment (julieahlquist) [reply cypress/integration/userCanLogout.spec.js:28]: yes I thought about that too but wasnt sure. I'll continue adding that

File: cypress/integration/userCanLogout.spec.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+describe('User Log-Out', () => {
+   beforeEach(() => {
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v0/articles",
+      response: "fixture:articles.json"
+    })
+  });
```
Comment (ghost) [inline cypress/integration/userCanLogout.spec.js:18]: General question, do  we need a PR with a request spec that verifies that logout has occurred?

Comment (sethnejame) [reply cypress/integration/userCanLogout.spec.js:18]: You can make one here to make sure the browser doesn't have anything in localStorage after the page is reloaded

**Review by sethnejame — APPROVED** [review]
Everything looks good to me, great work!

**Pull request merged**